DFE-Digital/get-help-to-retrain
# GET-783 Spell check service does not fail if no API key available

jira-ticket: https://dfedigital.atlassian.net/browse/GET-783

* Spell check service does not throw error when there is no API key
* Spell check service returns early if API key is not present
* Remove API key from test ENV, and only set in respective tests to stop making real calls in specs (which was happening previously)
* Fix tests in feature to avoid making real API calls when key set, also fix an old test as it will always pass otherwise

How to test:
tail the logs on heroku `heroku logs -t --app get-help-to-get-783-dis-ir3cm5` and perform a search, you should not see `Spell Check service error` in the logs

I will remove the bing keys from the vaults as we move along the pipeline
